Code/GraphMol/MonomerMol/AtomisticToMM.cpp

```diff
@@ -0,0 +1,216 @@
+#include "Conversions.h"
+
+#include <queue>
+#include <unordered_map>
+#include <memory>
+
+#include <GraphMol/RWMol.h>
+#include <GraphMol/ROMol.h>
+#include <GraphMol/MolOps.h>
+#include <GraphMol/MonomerInfo.h>
+#include <GraphMol/SmilesParse/SmilesParse.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <GraphMol/Substruct/SubstructMatch.h>
+
+#include "MonomerDatabase.h"
+#include "MonomerMol.h"
+
+namespace RDKit
+{
+
+namespace
+{
+
+const std::string ATTACH_FROM{"attachFrom"};
+const std::string MONOMER_IDX1{"monomerIndex1"};
+const std::string MONOMER_IDX2{"monomerIndex2"};
+const std::string REFERENCE_IDX{"referenceIndex"};
+
+// 3-letter to 1-letter amino acid code mapping
+// These are not currently in the monomer database, but some have
+// symbols that are already in the monomer database. We may need to
+// figure out how to have multiple 3-letter codes for a single symbol
+// polymer_type pair (Histidine is the best example)
+const std::unordered_map<std::string, char> backup_res_table({
```

Comment:
still little confused by why we need this after reading the docs. is this in place of a user-provided db?

Reply:
Yeah I will fix the doc here this is not well phrased.

Currently, monomer types and HELM symbol are the primary key to the monomer database and only 1 PDB code is allowed for each entry. This means we can't store multiple pronated states of monomers (for example, histidine) in the DB. This map allows commonly seen alternate 3-letter codes to be translated into HELM symbols

Reply:
ooh, I think this rings a bell. This you consider `std::multimap` or `std::unordered_multimap`? I feel like the snippet in https://en.cppreference.com/w/cpp/container/multimap/equal_range resembles your intended use case